commit 3b18325fcdb0eb9ad751f6e6cfe3ff03d191c837 (refs/changes/90/990/1)
Author: Chris Burroughs <chris.burroughs@joyent.com>
Date:   2016-11-30T13:56:05-05:00 (2 years, 10 months ago)
    
    ./bin/triton fwrule create -D "ssh" "FROM any TO all vms ALLOW tcp PORT 22"
    Created firewall rule fc030013-ecea-4b1f-8065-086e9003408e

diff --git a/lib/do_fwrule/do_create.js b/lib/do_fwrule/do_create.js
index d1ec831..2de8b87 100644
--- a/lib/do_fwrule/do_create.js
+++ b/lib/do_fwrule/do_create.js
@@ -45,15 +45,21 @@ function do_create(subcmd, opts, args, cb) {
         createOpts.description = opts.description;
     }
 
-    this.top.tritonapi.cloudapi.createFirewallRule(createOpts,
-            function (err, fwrule) {
-        if (err) {
-            cb(err);
-            return;
+    var tritonapi = this.top.tritonapi;
+    common.cliSetupTritonApi({cli: this.top}, function onSetup(setupErr) {
+        if (setupErr) {
+            cb(setupErr);
         }
-        console.log('Created firewall rule %s%s', fwrule.id,
-            (!fwrule.enabled ? ' (disabled)' : ''));
-        cb();
+        tritonapi.cloudapi.createFirewallRule(
+            createOpts, function (err, fwrule) {
+                if (err) {
+                    cb(err);
+                    return;
+                }
+                console.log('Created firewall rule %s%s', fwrule.id,
+                            (!fwrule.enabled ? ' (disabled)' : ''));
+                cb();
+            });
     });
 }
 
